### PR TITLE
arch/x86: Kconfig cleanup

### PR DESCRIPTION
**Pull request by ghost**
Remove options that are either unused or exactly equivalent to other options.

### CODE REVIEW

File: include/arch/x86/ffs.h [hunk 1]
```diff
@@ -47,7 +47,7 @@ static ALWAYS_INLINE unsigned int find_lsb_set(u32_t op)
 
 	__asm__ volatile (
 
-#if defined(CONFIG_CMOV)
+#if !defined(CONFIG_CPU_MINUTEIA)
```
Comment (ulfalizer) [inline include/arch/x86/ffs.h]: Might be clear enough anyway, but maybe a comment like `/* No CMOV */` could be added somewhere. Maybe the `#ifdef` could be reversed too.

**Review by ulfalizer — APPROVED** [review]
Looks great to me. Thanks!

File: include/arch/x86/ffs.h (hunk 1 above)
Comment (gnuless) [reply include/arch/x86/ffs.h]: Ah, good suggestions.  I've revised the last commit to take them into account.

**Review by nashif — APPROVED** [review]

File: include/arch/x86/ffs.h (hunk 1 above)
Comment (ulfalizer) [reply include/arch/x86/ffs.h]: Thx

**Pull request merged**